### PR TITLE
Pycommon actions - fixes and improvements

### PR DESCRIPTION
**Pull request by cejkato2**
Actions were improved, documentation (http://nemea.liberouter.org/reporting/) updated - it is still a draft but I've checked arguments of action types.

It is now possible to write advanced tests that can check what rules were performed and what actions were called.
Some `simulate` or `test` argument is now needed for actions in order to write tests of Mongo / Warden / Email. We want to check if they are called and generate a right content but we don't want to store it or send it anywhere..

Changes must be tested before releasing package.

### CODE REVIEW

File: pycommon/reporter_config/actions/Email.py [hunk 1]
```diff
@@ -4,45 +4,80 @@
...
+
+        self.addrFrom   = a["from"]     if "from"     in a else "nemea@localhost"
+        self.smtpServer = a["server"]   if "server"   in a else "localhost"
+        self.smtpPort   = a["port"]     if "port"     in a else 25
+        self.smtpTLS    = a["startTLS"] if "startTLS" in a else False
+        self.smtpSSL    = a["forceSSL"] if "forceSSL" in a else False
+        self.smtpUser   = a["authuser"] if "authuser" in a else None
+        self.smtpPass   = a["authpass"] if "authpass" in a else None
+        key    = a["key"]   if "key"   in a else None
+        chain  = a["chain"] if "chain" in a else None
```
Comment (vaclavbartos) [inline pycommon/reporter_config/actions/Email.py]: Do you know the `.get()` method? :-) For example `a.get("from", "nemea@localhost")` is much simpler.

File: pycommon/reporter_config/actions/Email.py [hunk 2]
```diff
@@ -4,45 +4,80 @@
...
 
-        self.smtp = smtplib.SMTP('localhost')
+        try:
+            with open(key, "r") as f:
+                self.smtpKey = f.read()
+        except Exception:
+            self.smtpKey = None
```
Comment (vaclavbartos) [inline pycommon/reporter_config/actions/Email.py]: It's not a good idea to skip error silently, an error should be logged somehow.

File: pycommon/reporter_config/actions/File.py [hunk 3]
```diff
@@ -17,18 +18,20 @@ class FileAction(Action):
...
+
+        self.path = a["path"] if "path" in a else None
+        if not self.path:
+            raise Exception("File action requires `path` argument.")
+
+        try:
+            # path already exists, check if it is a directory
+            s = os.stat(self.path)
+            self.dir = S_ISDIR(s.st_mode)
+        except:
```
Comment (vaclavbartos) [inline pycommon/reporter_config/actions/File.py]: Never use plain `except`, it also captures e.g. `KeyboardInterrupt`, use at least `except Exception` if it's not possible to use more specific exception.

File: pycommon/reporter_config/actions/File.py [hunk 4]
```diff
@@ -17,18 +18,20 @@ class FileAction(Action):
...
+            self.dir = False
 
         if self.path == '-':
             self.fileHandle = sys.stdout
```
Comment (vaclavbartos) [inline pycommon/reporter_config/actions/File.py]: (comment to line 38)
Why `dir=True` when input is stdout? It's confusing and I don't see any reason for it.

File: pycommon/reporter_config/actions/Email.py (hunk 1 above)
Comment (cejkato2) [inline pycommon/reporter_config/actions/Email.py]: Nice, I'll try it :-)

File: pycommon/reporter_config/actions/Email.py (hunk 2 above)
Comment (cejkato2) [inline pycommon/reporter_config/actions/Email.py]: Key is an optional argument in configuration so if it is no supplied, i want to skip the error from open(None, "r").
The same for cert file.

If you think we should log it, we can :-)

File: pycommon/reporter_config/actions/File.py (hunk 3 above)
Comment (cejkato2) [inline pycommon/reporter_config/actions/File.py]: Thanks, I'm changing it.

File: pycommon/reporter_config/actions/File.py (hunk 4 above)
Comment (cejkato2) [inline pycommon/reporter_config/actions/File.py]: I'm sorry, this is not my latest version of this file :-/ It looks different now. I'll push it soon.

File: pycommon/reporter_config/actions/Email.py (hunk 2 above)
Comment (vaclavbartos) [inline pycommon/reporter_config/actions/Email.py]: I agree it should be silently skipped if key=None. But if key is set but the file can't be opened (e.g. it doesn't exist), it should be reported.
So, there should be an `if` for the case key=None, but exceptions in file reading should be logged.

**Pull request merged**